tools/azure-devtools/src/azure_devtools/scenario_tests/preparers.py

```diff
@@ -59,18 +112,47 @@ def _prepare_create_resource(self, test_class_instance, **kwargs):
 
     def __call__(self, fn):
         def _preparer_wrapper(test_class_instance, **kwargs):
+            _logger.debug("Entering preparer wrapper for %s and test %s",
+                self.__class__.__name__, str(test_class_instance))
 
-            resource_name, kwargs = self._prepare_create_resource(test_class_instance, **kwargs)
+            resource_name, kwargs = self._prepare_create_resource(test_class_instance, fn, **kwargs)
 
             trim_kwargs_from_test_function(fn, kwargs)
 
             try:
+                test_class_instance._delay_removal = False # pylint: disable=protected-access
```

Comment:
Could you tell me how you use that?

Reply:
So this may not be relevant given your constraint of requiring parent elements to declare their cache status;  I'm inclined to keep it in however similar to my comment above of "if we ever go to that modality this suddenly becomes important again"

In that case it's used to "inform" parent classes that otherwise wouldn't have known they should be cached, that they shouldn't delete themselves until the end lest they nuke the cached sub-resources in that process.